auth/user_mgt.go

```diff
@@ -372,6 +373,82 @@ func processClaims(p map[string]interface{}) error {
 	return nil
 }
 
+// Error handlers.
+
+const (
+	emailAlredyExists        = "email-already-exists"
+	idTokenRevoked           = "id-token-revoked"
+	insufficientPermission   = "insufficient-permission"
+	phoneNumberAlreadyExists = "phone-number-already-exists"
+	projectNotFound          = "project-not-found"
+	uidAlreadyExists         = "uid-already-exists"
+	unknown                  = "unknown-error"
+	userNotFound             = "user-not-found"
+)
+
+// IsEmailAlreadyExists checks if the given error was due to a duplicate email.
+func IsEmailAlreadyExists(err error) bool {
+	return internal.HasErrorCode(err, emailAlredyExists)
+}
+
+// IsIDTokenRevoked checks if the given error was due to a revoked ID token.
+func IsIDTokenRevoked(err error) bool {
+	return internal.HasErrorCode(err, idTokenRevoked)
+}
+
+// IsInsufficientPermission checks if the given error was due to insufficient permissions.
+func IsInsufficientPermission(err error) bool {
+	return internal.HasErrorCode(err, insufficientPermission)
+}
+
+// IsPhoneNumberAlreadyExists checks if the given error was due to a duplicate phone number.
+func IsPhoneNumberAlreadyExists(err error) bool {
+	return internal.HasErrorCode(err, phoneNumberAlreadyExists)
+}
+
+// IsProjectNotFound checks if the given error was due to a non-existing project.
+func IsProjectNotFound(err error) bool {
+	return internal.HasErrorCode(err, projectNotFound)
+}
+
+// IsUIDAlreadyExists checks if the given error was due to a duplicate uid.
+func IsUIDAlreadyExists(err error) bool {
+	return internal.HasErrorCode(err, uidAlreadyExists)
+}
+
+// IsUnknown checks if the given error was due to a unknown server error.
+func IsUnknown(err error) bool {
+	return internal.HasErrorCode(err, unknown)
+}
+
+// IsUserNotFound checks if the given error was due to non-existing user.
+func IsUserNotFound(err error) bool {
+	return internal.HasErrorCode(err, userNotFound)
+}
+
+var serverError = map[string]string{
+	"CONFIGURATION_NOT_FOUND": projectNotFound,
+	"DUPLICATE_EMAIL":         emailAlredyExists,
+	"DUPLICATE_LOCAL_ID":      uidAlreadyExists,
+	"EMAIL_EXISTS":            emailAlredyExists,
+	"INSUFFICIENT_PERMISSION": insufficientPermission,
+	"PHONE_NUMBER_EXISTS":     phoneNumberAlreadyExists,
+	"PROJECT_NOT_FOUND":       projectNotFound,
+}
+
+func handleServerError(err error) *internal.FirebaseError {
```

Comment:
I would probably use `error` for the return type here, just because I've been bitten too many times by trying to use a specific error type.

Reply:
+1. Done